TheMovieDatabase/ViewControllers/Favorites/View/FavoritesViewController.swift

```diff
@@ -12,47 +12,84 @@ final class FavoritesViewController: UIViewController {
 
     // MARK: - Properties
 
-    private let service = MoviesLoadingService()
+    private let accountId = UserDefaults.standard.accountId
+    private let moviesLoadingService = MoviesLoadingService()
+    private let profileService = ProfileService()
     private var movies: [Movie] = []
+    private var wasShown = false
 
     // MARK: - Subviews
 
     @IBOutlet weak private var tableView: UITableView!
     @IBOutlet weak private var blankImage: UIImageView!
     @IBOutlet weak private var blankTitle: UILabel!
+    @IBOutlet weak private var activityIndicator: UIActivityIndicatorView!
 
     // MARK: - UIViewController
 
     override func viewDidLoad() {
         super.viewDidLoad()
         configureView()
+        loadFavoriteMovies()
     }
 
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(true)
-        movies = service.getFavoriteMovies()
-        if movies.isEmpty {
-            tableView.isHidden = true
-            blankImage.isHidden = false
-            blankTitle.isHidden = false
+        if wasShown {
+            movies = moviesLoadingService.getFavoriteMovies()
+            if movies.isEmpty {
+                setEmptyState(hidden: false)
+            } else {
+                setEmptyState(hidden: true)
+            }
+            tableView.reloadData()
         } else {
-            tableView.isHidden = false
-            blankImage.isHidden = true
-            blankTitle.isHidden = true
+            wasShown = true
         }
-        tableView.reloadData()
     }
 
     // MARK: - Private Methods
 
     private func configureView() {
+        activityIndicator.startAnimating()
+        setEmptyState(hidden: true)
         navigationController?.navigationBar.prefersLargeTitles = true
         tableView.delegate = self
         tableView.dataSource = self
         tableView.register(UINib(nibName: MovieTableViewCell.identifier, bundle: nil),
                            forCellReuseIdentifier: MovieTableViewCell.identifier)
         tableView.tableFooterView = UIView()
     }
+
+    private func loadFavoriteMovies() {
+        moviesLoadingService.loadFavoriteMovies(accountId: accountId) { [weak self] (movies) in
+            guard
+                let self = self,
+                let movies = movies
+            else {
+                return
+            }
+            self.movies = movies
+            self.tableView.reloadData()
+            self.updateView()
```

Comment:
Nit-pick: you don't need to use `guard let self = self` everywhere, it's only needed when you need non-optional self, for example, if you need to pass it to function

So there:
```
guard let movies = movies else {
    return
}
self?.movies = movies
self?.tableView.reloadData()
self?.updateView()
```

Btw, there is an unnecessary line break
```
guard let self = self, let movies = movies else {
    return
}
```
You don't need to add line breaks in every guard. Add only if it doesn't fit in one line